### PR TITLE
[shopsys] move some parameter attributes and filter functionality from project-base to framework and frontend-api bundles

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, the reason for the PR| We are slowly moving all functionality from project-base to appropriate packages. This is the next step.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://tl-luigis-box-fix-slider-parameters.odin.shopsys.cloud
  - https://cz.tl-luigis-box-fix-slider-parameters.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Migrations/Version20240613105832.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                ADD
+                    CONSTRAINT FK_69348FEF8BD700D FOREIGN KEY (unit_id) REFERENCES units (id) NOT DEFERRABLE INITIALLY IMMEDIATE');
+            $this->sql('CREATE INDEX IDX_69348FEF8BD700D ON parameters (unit_id)');
+        }
+
+        if ($this->isAppMigrationNotInstalledRemoveIfExists('Version20200910092545')) {
+            $this->sql('ALTER TABLE parameter_values ALTER text TYPE TEXT');
+        }
+
+        if (!$this->isAppMigrationNotInstalled('Version20200220124729')) {
```
Comment (grossmannmartin) [inline packages/framework/src/Migrations/Version20240613105832.php]: can we skip early returns in every migration file?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 2]
```diff
@@ -1123,6 +1123,15 @@ msgstr "Editace objednávky - č. %number%"
 msgid "Editing page"
 msgstr "Editace stránky"
 
+msgid "Editing parameter of type color"
+msgstr "Editace parametru typu barva"
+
+msgid "Editing parameter value of type color"
+msgstr "Editace hodnoty parametru typu barva"
+
+msgid "Editing pararameter value of type color - %name%"
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: Can you please fix the typo in msgid?

File: packages/frontend-api/src/Model/Product/Filter/ParameterFilterOption.php [hunk 3]
```diff
@@ -5,17 +5,33 @@
...
+     * @param bool $isCollapsed
+     * @param float|null $selectedValue
+     * @param bool|null $isSelectable
      */
     public function __construct(
         public readonly Parameter $parameter,
         public readonly array $values,
+        public bool $isCollapsed,
+        public ?float $selectedValue = null,
+        public ?bool $isSelectable = null,
```
Comment (grossmannmartin) [inline packages/frontend-api/src/Model/Product/Filter/ParameterFilterOption.php]: does it make sense to have this boolean nullable?

File: packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php [hunk 4]
```diff
@@ -14,21 +14,19 @@
 use Shopsys\FrameworkBundle\Model\Product\Flag\Flag;
 use Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter;
 use Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue;
+use Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue as BaseParameterValue;
```
Comment (grossmannmartin) [inline packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php]: This import should be unified with the one above

File: packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php [hunk 5]
```diff
@@ -64,27 +67,35 @@ protected function createBrandFilterOption(Brand $brand, int $count, bool $isAbs
...
+     * @param float|null $selectedValue
+     * @param bool|null $isSliderAllowed
      * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ParameterFilterOption
      */
     protected function createParameterFilterOption(
         Parameter $parameter,
         array $parameterValueFilterOptions,
+        bool $collapsed = false,
+        ?float $selectedValue = null,
+        ?bool $isSliderAllowed = null,
```
Comment (grossmannmartin) [inline packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php]: should be this bool nullable?

File: project-base/app/config/images.yaml [hunk 6]
```diff
@@ -51,3 +51,6 @@
     class: Shopsys\FrameworkBundle\Model\Seo\Page\SeoPage
     types:
         -   name: og
+
+-   name: parameterValue
```
Comment (grossmannmartin) [inline project-base/app/config/images.yaml:55]: How this was not necessary before and is now?

File: project-base/app/config/services.yaml [hunk 7]
```diff
@@ -253,9 +253,6 @@ services:
     Shopsys\FrameworkBundle\Component\Setting\Setting:
         alias: App\Component\Setting\Setting
 
-    Shopsys\FrameworkBundle\Component\UploadedFile\UploadedFileFacade:
-        alias: App\Component\UploadedFile\UploadedFileFacade
```
Comment (grossmannmartin) [inline project-base/app/config/services.yaml]: `App\Component\UploadedFile\UploadedFileFacade` still exists, so the alias should remain, or?

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: project-base/app/config/images.yaml (hunk 6 above)
Comment (TomasLudvik) [reply project-base/app/config/images.yaml:55]: Before was this functionality done different way, I have removed custom code and replaced it with generic one.

Comment (TomasLudvik) [reply project-base/app/config/images.yaml:55]: https://github.com/shopsys/shopsys/blob/409a4702d90666344b0350c2d0cdf464aec5337b/project-base/app/src/Model/Product/Parameter/ParameterFacade.php#L137

vs

https://github.com/shopsys/shopsys/blob/c2685e96d25523e57b0db5a037d44ce73e3b5772/packages/framework/src/Model/Product/Parameter/ParameterFacade.php#L217

File: project-base/app/config/services.yaml (hunk 7 above)
Comment (TomasLudvik) [reply project-base/app/config/services.yaml]: Good catch, thx!

File: packages/frontend-api/src/Model/Product/Filter/ParameterFilterOption.php (hunk 3 above)
Comment (TomasLudvik) [reply packages/frontend-api/src/Model/Product/Filter/ParameterFilterOption.php]: I have changed order and nullability of the parameters

File: packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php (hunk 5 above)
Comment (TomasLudvik) [reply packages/frontend-api/src/Model/Product/Filter/ProductFilterOptionsFactory.php]: I have changed order and nullability of the parameters

**Review by grossmannmartin — APPROVED** [review]

**Pull request merged**